validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/ProduceBlockRequest.java

```diff
@@ -0,0 +1,235 @@
+import tech.pegasys.teku.infrastructure.json.types.DeserializableOneOfTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.datastructures.blocks.BlockContainer;
+import tech.pegasys.teku.spec.datastructures.blocks.BlockContainerSchema;
+import tech.pegasys.teku.validator.remote.typedef.ResponseHandler;
+
+public class ProduceBlockRequest extends AbstractTypeDefRequest {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final UInt64 slot;
+  private final boolean preferSszBlockEncoding;
+  private final BlockContainerSchema<BlockContainer> blockContainerSchema;
+  private final BlockContainerSchema<BlockContainer> blindedBlockContainerSchema;
+  private final DeserializableTypeDefinition<ProduceBlockResponse> produceBlockResponseDefinition;
+  private final DeserializableTypeDefinition<ProduceBlockResponse>
+      produceBlindedBlockResponseDefinition;
+  private final ResponseHandler<ProduceBlockResponse> responseHandler;
+
+  public final DeserializableOneOfTypeDefinition<ProduceBlockResponse> produceBlockTypeDefinition;
+
+  public ProduceBlockRequest(
+      final HttpUrl baseEndpoint,
+      final OkHttpClient okHttpClient,
+      final Spec spec,
+      final UInt64 slot,
+      final boolean preferSszBlockEncoding) {
+    super(baseEndpoint, okHttpClient);
+    this.slot = slot;
+    this.preferSszBlockEncoding = preferSszBlockEncoding;
+    blockContainerSchema = spec.atSlot(slot).getSchemaDefinitions().getBlockContainerSchema();
+    blindedBlockContainerSchema =
+        spec.atSlot(slot).getSchemaDefinitions().getBlindedBlockContainerSchema();
+
+    produceBlockResponseDefinition =
+        DeserializableTypeDefinition.object(ProduceBlockResponse.class)
+            .initializer(ProduceBlockResponse::new)
+            .withField(
+                EXECUTION_PAYLOAD_BLINDED,
+                BOOLEAN_TYPE,
+                ProduceBlockResponse::getExecutionPayloadBlinded,
+                ProduceBlockResponse::setExecutionPayloadBlinded)
+            .withField(
+                EXECUTION_PAYLOAD_VALUE,
+                UINT256_TYPE,
+                ProduceBlockResponse::getExecutionPayloadValue,
+                ProduceBlockResponse::setExecutionPayloadValue)
+            .withField(
+                "data",
+                blockContainerSchema.getJsonTypeDefinition(),
+                ProduceBlockResponse::getData,
+                ProduceBlockResponse::setData)
+            .withField(
+                "version",
+                DeserializableTypeDefinition.enumOf(SpecMilestone.class),
+                ProduceBlockResponse::getSpecMilestone,
+                ProduceBlockResponse::setSpecMilestone)
+            .build();
+
+    produceBlindedBlockResponseDefinition =
```

Comment:
Done, also moved the `produceBlockResponseDefinition` and `produceBlindedBlockResponseDefinition` to local variables within the constructor since they're used there only